### DISCUSSION TITLE
Make project as a starter

### DISCUSSION THREAD

**Discussion by PittyXu** [9tigerio/db2rest#505] [Ideas]
This is a fantastic project, I think it can extract modules to make the project as a starter, which can integrate it into existing projects.

It also can support gradle build for multiple modules.

**Comment by kdhrubo** [comment]
Will you be able to help in doing that ?

**Reply by PittyXu** [reply]
I make a PR. #507

**Reply by kdhrubo** [reply]
@PittyXu 
I did review the PR. Looks good. But we would prefer to stick to Maven instead of Gradle as like me other developers working on the project are not familiar with Gradle. Is it possible to create Maven multi module project with a starter please? Everything including release is working fine with Maven and we intend not to break it. 

Also thinking will it be good to make 2 starters - one for relational database and other for mongo and bring them into the main db2rest project. That way any spring boot project can use it in future (we do need to make some adjustments to our service interfaces though)

**Reply by PittyXu** [reply]
I'm not familiar with Maven 😭

**Reply by thadguidry** [reply]
@PittyXu it's actually not hard at all.  You can do it!  Here is a great [Baeldung guide on multi-module projects with Maven](https://www.baeldung.com/maven-multi-module).
I would also encourage us to switch to Maven 4 which allows us greater granularity on scope with compiling submodules and dependencies.  Here's a [Guide to Working with Multi Modules in Maven 4](https://maven.apache.org/guides/mini/guide-multiple-modules-4.html).
